language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/CodeActionContext.java

```diff
@@ -60,9 +62,16 @@ public interface CodeActionContext extends DocumentServiceContext {
     int cursorPositionInTree();
 
     /**
-     * Get the node at cursor.
+     * Get the node at selected range.
      *
-     * @return {@link Integer} offset of the cursor
+     * @return {@link Node}
+     */
+    Node nodeAtRange();
```

Comment:
Let's deprecate the `nodeAtCursor()` method and add a new `nodeAtRange()` method. Similarly what we did for `cursorPosition()`. Other changes are ok.